mm2src/coins/eth.rs

```diff
@@ -1646,18 +1672,27 @@ impl EthCoin {
     }
 
     fn approve(&self, spender: Address, amount: U256) -> EthTxFut {
```

Comment:
I have remembered one thing: could you please call the `approve` with `amount = U256::max`? Most ETH/BNB dApps, that I've tried, use the same approach. It saves some gas by avoiding repetitive approval calls.